src/core/paths.ts

```diff
@@ -7,10 +7,10 @@ import { Uri } from "vscode";
 import { isUriString } from "./uri";
 
 export function isFileSystemLocation(file: string | Uri | URL) {
-    if (file instanceof Uri) return file.scheme === "file";
-    if (file instanceof URL) return file.protocol === "file:";
+    if (file instanceof Uri) return file.scheme === "file" || file.scheme === "vscode-remote";
+    if (file instanceof URL) return file.protocol === "file:" || file.protocol === "vscode-remote:";
     if (typeof file !== "string") return false;
-    if (isUriString(file)) return file.startsWith("file://");
+    if (isUriString(file)) return file.startsWith("file://") || file.startsWith("vscode-remote://");

```

Comment:
Can't any URI be readable via the FS since that's what VS Code's FS does? or is this intended to specifically handle "remote" cases where there actually is a real file backing it?